app/controllers/application_controller.rb

```diff
@@ -1,2 +1,21 @@
 class ApplicationController < ActionController::Base
+    add_flash_types  :success
+    
+    helper_method :current_user, :logged_in?
+
+    def current_user
+        @current_user ||= User.find_by_id(session[:user_id])
+
+    end
+    
+    def logged_in?
+        current_user != nil
+    end
+    
+    def require_user
+        if !logged_in?
+            flash[:danger] = "You must be logged in to perform that action."
+            redirect_to login_path
+        end
```

Comment:
Another nice way to cleanup your code, which `rubocop` would warn you about is to only use `if ..... end` if there is more then 1 outcome of the `if`. Here we have only 1 outcome, so we could do what is called a 'guard clause'.

```
def require_user
        return if logged_in?
        flash[:danger] = "You must be logged in to perform that action."
        redirect_to login_path
end
```

Now we have less lines, less indentation and we dont use a a not check `!logged_in?`, its easier to read `if logged_in?`

Reply:
(optional)